### PR TITLE
cask-repair revamp

### PR DESCRIPTION
**Pull request by vitorgalvao**
Revamp of `cask-repair`, with some important changes:

+ [Way easier to setup and use](https://github.com/caskroom/homebrew-cask/pull/29079):
  + No longer needs any manual forking or setting of remotes.
  + No need to fiddle with `--push|-p` and `--pull|-l` flags (they were deprecated).
  + No longer need to be inside a `homebrew-*/Casks` repo. Works from anywhere.
+ You can now give multiple cask arguments at once.
+ You can now give `s` as the version to skip a cask.
+ Added `--install` flag to install updated cask right after submitting PR.
+ Auto-skip casks with `language`.
+ Introduces new method of `appcast`-checking (https://github.com/caskroom/homebrew-cask/pull/28966).
+ Fixes bugs (and may introduce others).

A few relevant implementation details (the script will visibly warn you with bright text when doing these actions):

+ All main repos are auto-tapped. I find this an acceptable tradeoff to being able to update any main cask from any directory.
+ A new `cask-repair` remote is created, pointing to your own repo. This does not take extra space (it’s just a reference) and facilitates a bunch of actions, including (but not limited to) initial setup.

For the future:

+ I’d like to make this work as `brew cask repair <cask>` instead of `cask-repair <cask>`. The feasibility of that depends on https://github.com/caskroom/homebrew-cask/issues/28977.

After merging this:

+ [ ] Update [formula](https://github.com/vitorgalvao/homebrew-tiny-scripts/blob/master/cask-repair.rb).
+ [ ] https://github.com/caskroom/homebrew-cask/pull/29079
+ [ ] https://github.com/caskroom/homebrew-cask/pull/28966
  + [ ] https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cask/lib/hbc/audit.rb

---

Pinging some heavy users of `cask-repair`: @adidalal @miccal @victorpopkov @artdevjs.

### CODE REVIEW

**Comment by yurikoles** [comment]
FINALLY! 
@vitorgalvao thanks for you work 👍

**Comment by adidalal** [comment]
Looks 👍 

Seems like it auto-taps all other -caskroom repos, but that's a minor caveat in exxhange for all the automation.

A sample run:

```bash
[~/Downloads] > ./cask-repair.sh calibre
`caskroom/homebrew-versions` not tapped. Tapping…
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
No changes to formulae.

==> Tapping caskroom/versions
Cloning into '/usr/local/Homebrew/Library/Taps/caskroom/homebrew-versions'...
remote: Counting objects: 219, done.
remote: Compressing objects: 100% (217/217), done.
remote: Total 219 (delta 24), reused 24 (delta 0), pack-reused 0
Receiving objects: 100% (219/219), 80.08 KiB | 0 bytes/s, done.
Resolving deltas: 100% (24/24), done.
Tapped 0 formulae (237 files, 288.4K)
`caskroom/homebrew-fonts` not tapped. Tapping…
==> Tapping caskroom/fonts
Cloning into '/usr/local/Homebrew/Library/Taps/caskroom/homebrew-fonts'...
remote: Counting objects: 1069, done.
remote: Compressing objects: 100% (652/652), done.
remote: Total 1069 (delta 695), reused 477 (delta 414), pack-reused 0
Receiving objects: 100% (1069/1069), 196.12 KiB | 0 bytes/s, done.
Resolving deltas: 100% (695/695), done.
Tapped 0 formulae (1,085 files, 863.4K)
`caskroom/homebrew-eid` not tapped. Tapping…
==> Tapping caskroom/eid
Cloning into '/usr/local/Homebrew/Library/Taps/caskroom/homebrew-eid'...
remote: Counting objects: 24, done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 24 (delta 0), reused 8 (delta 0), pack-reused 0
Unpacking objects: 100% (24/24), done.
Tapped 0 formulae (64 files, 45K)
Updating taps… Already up-to-date.
A `cask-repair` remote does not exist. Creating it now…
--------------------------------------------------------------------------------
cask 'calibre' do
<snip>
Type the new version (or leave blank to use current one, or use `s` to skip)
```

**Comment by vitorgalvao** [comment]
@adidalal Those are very much expected. Added that info to the top post:

> A few relevant implementation details (the script will visibly warn you with bright text when doing these actions):
>
> + All main repos are auto-tapped. I find this an acceptable tradeoff to being able to update any main cask from any directory.
> + A new `cask-repair` remote is created, pointing to your own repo. This does not take extra space (it’s just a reference) and facilitates a bunch of actions, including (but not limited to) initial setup.

**Comment by yurikoles** [comment]
What to do with ambiguity in homebrew-versions name? For example I use it for 
homebrew's fork and cask-versions for cask's fork.

**Comment by vitorgalvao** [comment]
@yurikoles There’s no ambiguity. This only acts on caskroom repos.

**Comment by alyssais** [comment]
Is it still possible to use a non-tap copy of the repository? I like to keep my working copies separate from my system Homebrew installation.

**Comment by yurikoles** [comment]
```
Error creating fork: yurikoles/homebrew-versions already exists on github.com
```
It does exist and its a fork from homebrew, not from cask.

**Comment by vitorgalvao** [comment]
> Is it still possible to use a non-tap copy of the repository?

No. This doesn’t significantly affect your system installation, though. Only thing it does to it is add a new remote, which was always part of the instructions anyway. Like yourself, I did to use a separate directory, but it simply isn’t worth it. `hub` even expects `origin` to be the main repository, and not having it setup like that caused a bunch of issues in the past.

> It does exist and its a fork from homebrew, not from cask.

Ah, I see your problem. That is not something I can fix, as the forking is done by `hub`. It is also something I *will not fix*, because you should no longer have that fork anyway, as that repo [has been deprecated](https://github.com/Homebrew/homebrew-versions#homebrewversions-deprecated). We don’t share any other names with other homebrew repos, so this won’t be a problem.

**Comment by yurikoles** [comment]
```
$ git remote -v
cask-repair	https://github.com//homebrew-cask (fetch)
cask-repair	https://github.com//homebrew-cask (push)
origin	https://github.com/caskroom/homebrew-cask (fetch)
origin	https://github.com/caskroom/homebrew-cask (push)
```
Hmm...

**Comment by vitorgalvao** [comment]
@yurikoles 

1. Does `grep 'oauth_token:' "${HOME}/.config/hub" 2>/dev/null` show you a token (no need to post it)?
2. What’s the output of `hub config --get github.user`?

**Comment by yurikoles** [comment]
1. There is a token.
2. Nothing.

Hub fork can create forks within `yurikoles/*` by invoking hub fork within repo.

**Comment by vitorgalvao** [comment]
@yurikoles What about `cat ~/.config/hub`? Feel free to omit the value of the token, but the rest is needed.

**Comment by miccal** [comment]
Looks fantastic @vitorgalvao, great work!

Silly question, but now `hub` is not required?

**Comment by yurikoles** [comment]
```
github.com:
- user: yurikoles
  oauth_token:  <my_token>
  protocol: https
```

**Comment by adidalal** [comment]
Hub gets called automatically. You will still need it installed

**Comment by miccal** [comment]
Thanks @adidalal, I just saw that in the code!

**Comment by vitorgalvao** [comment]
@yurikoles Should work now.

File: cask-repair [hunk 1]
```diff
@@ -360,6 +360,9 @@ while [[ "${1}" ]]; do
       can_run_without_arguments='true'
       delete_created_branches='true'
       ;;
+    -p | -l | --pull | --push)
+      failure_message '--push and --pull have been deprecated. They no longer do anything.\nYou also no longer need to be inside a hombrew-*/Casks directory.\n\nFor a list of all changes, see https://github.com/vitorgalvao/tiny-scripts/pull/56'
```
Comment (Numbermaniac) [inline cask-repair]: Small typo:

> You also no longer need to be inside a **hombrew**-*/Casks directory.

Comment (vitorgalvao) [inline cask-repair]: Fixed.

**Comment by vitorgalvao** [comment]
This is now released, and the stable version.

**Pull request merged**